File: Algorithms/347/top-k-frequent-elements.swift
class Solution {
    func topKFrequent(_ nums: [Int], _ k: Int) -> [Int] {
        var ary = Array(Set(nums))
        var cntDic = [Int: Int]()
        
        for num in nums {
            cntDic[num] = (cntDic[num] ?? 0) + 1
        }
        ary.sort {
            (cntDic[$0] ?? 0) > (cntDic[$1] ?? 0)
        }
        
        return Array(ary[0..<k])
    }
}

File: Algorithms/[1, 1k)/347/top-k-frequent-elements.cpp
map<int, int> cntMap;
bool cmp(int a, int b) {
    return cntMap[a] > cntMap[b];
}

class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        cntMap.clear();
        for (auto num : nums) cntMap[num] += 1;
        
        sort(nums.begin(), nums.end());
        nums.erase(unique(nums.begin(), nums.end()), nums.end());
        
        sort(nums.begin(), nums.end(), cmp);
        while (nums.size() > k) nums.pop_back();
        return nums;
    }
};